server/src/main/java/org/elasticsearch/search/aggregations/metrics/MinAggregator.java

```diff
@@ -219,6 +222,9 @@ public void visit(int docID) {
 
                 @Override
                 public void visit(int docID, byte[] packedValue) {
+                    if (lookupCounter.incrementAndGet() > MAX_BKD_LOOKUPS) {
```

Comment:
I think if we exit early, we need to set result back to `null` to signal that we didn't find a value?

Reply:
I may be wrong, but it looks that the only possibility where returned `result[0]` is not `null` is the case when the `liveDocs.get(docID)` is `true`, so when the intersection process is terminated by early exit the returned value is `null`.

Reply:
Ah sorry, I was looking at the line where `result = new Number[1];` and missed that the return value is `result[0]` and not just `result`.  All good :+1: